test/cli/warm-cache/test.out

```diff
@@ -0,0 +1,14 @@
+====first run (cold cache)====
+                         types.input.files :              3
+            types.input.files.kvstore.miss :              3
+====second run (warm cache)====
+                         types.input.files :              3
+             types.input.files.kvstore.hit :              3
+====first after change (one cache miss)====
+                         types.input.files :              3
+             types.input.files.kvstore.hit :              2
+            types.input.files.kvstore.miss :              1
+====second after change (warm cache)====
+                         types.input.files :              3
+             types.input.files.kvstore.hit :              2
+            types.input.files.kvstore.miss :              1
```

Comment:
This should be 

```suggestion
             types.input.files.kvstore.hit :              3
```

and I'm not sure why it's not. But at least we don't crash now so this is progress?

Reply:
This turned out to be a separate issue: we only write the trees to the cache if GlobalState changed, which mean that adding a comment to a file will cause a cache miss but prevent that file from getting written back into the cache.